### PR TITLE
feature/US10 + US14: Visualize and edit your profile, visualize other user's profile and follow them!

### LINKED ISSUE #155: US10-FE#1 Display user info

### LINKED ISSUE #156: US10-FE#2 Edit user info
A profile picture may be added/changed from edit user info.

### LINKED ISSUE #157: US10-FE#3 Add follows and followers section
These sections are not part of user info

### LINKED ISSUE #158: US10-FE#4 Add created recipes section
Also add the option to modify or delete each recipe.

These sections are not part of user info

### LINKED ISSUE #169: US14-FE#1 Add follow option in recipe details

### LINKED ISSUE #171: US14-FE#3 Add follows and followers lists
This issue is similar to #157

### LINKED ISSUE #172: US14-FE#4 Show User Recipes on follow/follower click

### LINKED ISSUE #176: feature/user-profile-visualization

### PR DESCRIPTION
**Pull request by guzmanalejandro**
At the moment, the visualization and editing (some functional details remain) of the user profile has been added, as a new component, pending to be added to the navigation bar for easy access by the user.

The user's profile is displayed with their profile image, with the option to modify or delete it.

Screenshot 2023-11-17 at 17 11 46

Pressing the edit user profile button allows you to edit the text fields and update them in the database as long as there are no errors on its part, such as a repeated email.

Screenshot 2023-11-17 at 10 20 35

### CODE REVIEW

**Comment by ivanmansilla** [comment]
For now, I think that what has been implemented is correct, since the functionality as you say is fine. It is the first version, so surely in the near future the design will be better, as a recommendation we could put that has a background with some existing color on the web, that's something that already commented the PO, for my part you can do the merge if it does not give error.

**Review by ivanmansilla — DISMISSED** [review]

**Comment by guzmanalejandro** [comment]
**UPDATE 18/11**
Now a distinction will be made between your own user and the user profile of another person, showing the recipes of the specific user and allowing you to edit, obviously and only, your user.

At the request of the product owner, the appearance of the user profile has been changed, and the fact of activating the admin mode only with your user has been better controlled.

Screenshot 2023-11-19 at 12 07 37

File: src/components/UserProfile.jsx [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+  const ImageRemoveQuestionClose = (event) => {
+    setRemoveQuestion(false)
+  };
+
+  const handleConfirmationClose = () => {
+    setShowConfirmation(false);
+  };
+
+  const getRecipes = () => {
+    fetch(`http://127.0.0.1:8000/recipe/user/${userName}`, {
```
Comment (andreu-vall) [inline src/components/UserProfile.jsx]: You should use the enironment variable with the backend address to also work when deployed

**Review by andreu-vall — DISMISSED** [review]
I did not try it as I am in the phone right now. Looking at it fast it looks correct but you should use the environment variable REACT_APP_API_URL for the backend address, because hardcoded to local will not work when deployed because the backend will not be local then, as it is another service with a custom url.

**Comment by andreu-vall** [comment]
And when developing in local, if you do not want to change the .env with the local address (because this file is tracked), you may create a .env.local file that has the same contents but with your desired url version, and it will override the .env variables without having to change the .env.

File: src/components/UserProfile.jsx (hunk 1 above)
Comment (guzmanalejandro) [reply src/components/UserProfile.jsx]: I know it, when I mark as complete the US I will use the env variable.

Comment (andreu-vall) [reply src/components/UserProfile.jsx]: 👍

**Comment by guzmanalejandro** [comment]
**UPDATE 23/11**
US14 implemented, with the option to follow users and visit their profiles, being able to see the information of said profile and the recipes that the user has uploaded (as long as the user has a public profile). In addition to deleting a user's recipes, the option to edit a recipe has been delayed to be implemented in the next sprint.

- Here you can see the profile of the user who is using the website, being able to edit their profile and see their recipes.
Screenshot 2023-11-23 at 17 52 51

- On the other hand, when visiting another user's profile, you can follow them in addition to looking at the public recipes they have uploaded to the website.
Screenshot 2023-11-23 at 17 53 02

**Comment by xvxnoah** [comment]
> **UPDATE 23/11** US14 implemented, with the option to follow users and visit their profiles, being able to see the information of said profile and the recipes that the user has uploaded (as long as the user has a public profile). In addition to deleting a user's recipes, the option to edit a recipe has been delayed to be implemented in the next sprint.
> 
> * Here you can see the profile of the user who is using the website, being able to edit their profile and see their recipes.
> 
> Screenshot 2023-11-23 at 17 52 51
> * On the other hand, when visiting another user's profile, you can follow them in addition to looking at the public recipes they have uploaded to the website.
> 
> Screenshot 2023-11-23 at 17 53 02

Impressive job Alejandro! I will list here a couple of things to consider that I have found while checking your US:

1. You are still using hardcoded URL ('http://127.0.0.1:8080'). You should use the URL of the backend deployment, which makes it easier to test everything.
2. You should solve the conflicts with the main branch in order to be able to merge without problems.
3. The recipe container of the image is touching with the bottom of the profile container.
4. When opening the list of followers or followings, if there are none to show, we should show a message.
5. The image of the user seems a bit flattened, we should check that.
6. When trying to change the user image, it should be done with just one click on the edit pencil, not 2 clicks.
7. There is a package from npm that is causing some problems, you should check that.
8. If the user has set a profile image, if would be nice to show it next to its username in the navbar; @BlessedLongsword this might be for you to check if you have time.

**Review by xvxnoah — APPROVED** [review]

**Pull request merged**